ml4gw/waveforms/taylorf2.py

```diff
@@ -0,0 +1,198 @@
+    pfaN = 3.0 / (128.0 * eta)
+    pfa_v0 = 1.0
+    pfa_v0 *= pfaN
+    pfa_v1 = 0.0
+    pfa_v1 *= pfaN
+    pfa_v2 = 5.0 * (74.3 / 8.4 + 11.0 * eta) / 9.0
+    pfa_v2 *= pfaN
+    pfa_v3 = -16.0 * PI
+    pfa_v3 *= pfaN
+    pfa_v4 = (
+        5.0
+        * (3058.673 / 7.056 + 5429.0 / 7.0 * eta + 617.0 * eta * eta)
+        / 72.0
+    )
+    pfa_v4 *= pfaN
+    pfa_v5logv = 5.0 / 3.0 * (772.9 / 8.4 - 13.0 * eta) * PI
+    pfa_v5logv *= pfaN
+    pfa_v5 = 5.0 / 9.0 * (772.9 / 8.4 - 13.0 * eta) * PI
+    pfa_v5 *= pfaN
+    pfa_v6logv = -684.8 / 2.1
+    pfa_v6 = (
+        11583.231236531 / 4.694215680
+        - 640.0 / 3.0 * PI * PI
+        - 684.8 / 2.1 * GAMMA
+        + eta * (-15737.765635 / 3.048192 + 225.5 / 1.2 * PI * PI)
+        + eta * eta * 76.055 / 1.728
+        - eta * eta * eta * 127.825 / 1.296
+        + pfa_v6logv * torch.log(torch.tensor(4.0))
+    )
+    pfa_v6logv *= pfaN
+    pfa_v6 *= pfaN
+    pfa_v7 = PI * (
+        770.96675 / 2.54016 + 378.515 / 1.512 * eta - 740.45 / 7.56 * eta * eta
+    )
+    pfa_v7 *= pfaN
+    # construct power series
+    phasing = (v7.T * pfa_v7).T
+    phasing += (v6.T * pfa_v6 + v6_logv.T * pfa_v6logv).T
+    phasing += (v5.T * pfa_v5 + v5_logv.T * pfa_v5logv).T
+    phasing += (v4.T * pfa_v4).T
+    phasing += (v3.T * pfa_v3).T
+    phasing += (v2.T * pfa_v2).T
+    phasing += (v1.T * pfa_v1).T
+    phasing += (v0.T * pfa_v0).T
+    # Divide by 0PN v-dependence
+    phasing /= v5
+
+    return phasing
+
+
+def taylorf2_amplitude(f: TensorType, mass1, mass2, distance) -> TensorType:
+    # Energy and Flux PN coeffiecients
+    mass1_s = mass1 * MTSUN_SI
+    mass2_s = mass2 * MTSUN_SI
+    M_s = mass1_s + mass2_s
+    eta = mass1_s * mass2_s / M_s / M_s
+    Mf = (f.T * M_s).T
+    v = (PI * Mf) ** (1.0 / 3.0)
+    v10 = v**10
+
+    FTaN = 32.0 * eta * eta / 5.0
+    FTa2 = -(12.47 / 3.36 + 3.5 / 1.2 * eta)  # noqa  F841
+    FTa3 = 4.0 * PI  # noqa  F841
+    FTa4 = -(  # noqa  F841
+        44.711 / 9.072
+        - 92.71 / 5.04 * eta
+        - 6.5 / 1.8 * eta * eta  # noqa  F841
+    )  # noqa  F841
+    FTa5 = -(81.91 / 6.72 + 58.3 / 2.4 * eta) * PI  # noqa  F841
+    FTa6 = (  # noqa  F841
+        664.3739519 / 6.9854400
+        + 16.0 / 3.0 * PI * PI
+        - 17.12 / 1.05 * GAMMA
+        - 17.12 / 1.05 * torch.log(torch.tensor(4.0))
+        + (4.1 / 4.8 * PI * PI - 134.543 / 7.776) * eta
+        - 94.403 / 3.024 * eta * eta
+        - 7.75 / 3.24 * eta * eta * eta
+    )  # noqa
+    FTa7 = (  # noqa  F841
+        -(162.85 / 5.04 - 214.745 / 1.728 * eta - 193.385 / 3.024 * eta * eta)
+        * PI
+    )
+
+    # Energy coefficients
+    dETaN = 2 * (-eta / 2.0)
+    dETa1 = 2.0 * -(0.75 + eta / 12.0)  # noqa  F841
+    dETa2 = 3.0 * -(  # noqa  F841
+        27.0 / 8.0 - 19.0 / 8.0 * eta + 1.0 / 24.0 * eta * eta
+    )  # noqa  F841
+    dETa3 = 4.0 * -(  # noqa  F841
```

Comment:
Why do all the `FTa#`s and `dETa#`s not get used anywhere?

Reply:
That's because taylorf2 is newtonian order in amplitude. I'll remove them now, we can bring them back when we get phenomD in.